File: Controllers/Resources/ProductResource.cs
using System;
using System.Collections.Generic;
using pc_store.Core.Models;

namespace pc_store.Controllers.Resources
{
    public class ProductResource : KeyValuePairResource
    {
        public string Description { get; set; }
        public decimal Price { get; set; }
        public int Stock { get; set; }
        public CategoryResource Category { get; set; }
        public SubCategoryResource SubCategory { get; set; }
        public ICollection<ProductPhoto> ProductPhotos { get; set; }
        public ICollection<SpecificationPhoto> SpecificationPhotos { get; set; }
        public string VideoLink { get; set; }
        public DateTime Created { get; set; }
        public DateTime Modified { get; set; }
    }
}

File: ClientApp/app/shared/models/product-resource.ts
import { SubCategory } from './sub-category';
import { Category } from './category';
import { ProductPhoto } from './product-photo';
import { SpecificationPhoto } from './specification-photo';

export interface ProductResource
{
    id: number;
    name: string;
    description: string;
    price: number;
    stock: number;
    productPhotos?: ProductPhoto[];
    specificationPhotos?: SpecificationPhoto[];
    category: Category;
    subCategory: SubCategory;
    videoLink: string;
    created?: Date;
    modified?: Date;
}



